🆕🎨 Design updates IV

fairlighteth opened the pull request:
# Summary

- Restyle and fixes for the token table
- Click on token -> links to SELL token selected on the SWAP container
- Changed ACTIONS: Replace BUY/SELL with Etherscan icon (VIEW CONTRACT) + APPROVE status. On click of the token name itself it will go to the SWAP page, selecting the token as a SELL token. This to simplify the token table UI.
- Mobile optimize (for now horizontal scroll)

<img width="1254" alt="Screen Shot 2022-11-18 at 13 13 23" src="https://user-images.githubusercontent.com/31534717/202713301-6589d698-4148-44a8-bb07-50d8cf183636.png">
<img width="1198" alt="Screen Shot 2022-11-18 at 13 13 36" src="https://user-images.githubusercontent.com/31534717/202713337-0912042a-a29e-4851-84e2-28206f2ba6cf.png">

<img width="388" alt="Screen Shot 2022-11-18 at 13 14 00" src="https://user-images.githubusercontent.com/31534717/202713393-4a63ad96-f8e1-4fc4-9647-21dc76d17220.png">

elena-zh commented:
Hey @fairlighteth , great changes!!
Some issues: 
1. The page content is truncated when screen width is in [960-1090]px
![reduced](https://user-images.githubusercontent.com/70885163/202731575-9a167c31-287b-4308-8be7-aaf0ea4cb149.png)
2. Mobile: lists dropdown is truncated
![image](https://user-images.githubusercontent.com/70885163/202731719-3cec8944-abb4-48cb-b5d0-166d27e2dc37.png)
3. Mobile: Search area is mostly hidden on Favourites list
![image](https://user-images.githubusercontent.com/70885163/202731953-c819d89c-68b3-48b4-87d9-0667bf5cc52a.png)
4. It would be great not to truncate a placeholder in the search field:
![image](https://user-images.githubusercontent.com/70885163/202732094-f035e428-cf90-4404-8bbe-4fc1b5ef9dba.png)

5. It would be great to expand a message area when no search results
![image](https://user-images.githubusercontent.com/70885163/202732543-18ff85fb-70fc-4374-87bb-6efae9a6155d.png)

Thanks!

fairlighteth commented:
@elena-zh thanks for reviewing. Addressed all points except:
1) Let's address this separately 
4) Also let's address this separately. We could show a shorter text for mobile.

shoom3301 reviewed (approved):
LGTM!

elena-zh commented:
Hey @fairlighteth , 
I have created #1444 issue for the 1, 4 cases
cases 5-6 LGTM!

As for the rest ones: the table does not look good now on a reduced screen: 
![image](https://user-images.githubusercontent.com/70885163/203036469-883a44fb-10c5-4b5a-8da6-f00df79d0e7e.png)

https://user-images.githubusercontent.com/70885163/203036690-c962b47d-fc02-4abb-98fb-906a9d9528f2.mp4


Could you please fix it? 
Thanks!

fairlighteth commented:
@elena-zh Pushed some updates. For resolutions around 960 <-> 1024px there are still some issues, which I like to address later.

elena-zh commented:
Hey @fairlighteth , great job!
One nitpick is that 'hovered' effect looks like truncated line on reduced screens, and when the table is scrolled. 
![hover is truncated](https://user-images.githubusercontent.com/70885163/203790409-d966f69e-50a1-44ea-be9b-797f8b8497d8.png)

Should I open a separate issue for this?

anxolin commented on src/custom/components/Tokens/TokensTableRow.tsx:
```diff
@@ -189,19 +185,12 @@ const DataRow = ({
-          <TableButton color={theme.red1}>Sell</TableButton>
-        </Link>
+        <ExtLink href={getBlockExplorerUrl(chainId, tokenData.address, 'token')}>
+          <TableButton>Contract</TableButton>
```
I find this a bit strange

I would actually expect the opposite, click on the token would take me to Etherscan, and the link would read as "Trade".

not super strong opinion though

elena-zh reviewed (approved):
Approved with #1496 issue

fairlighteth commented:
@elena-zh Can you check the latest? I fixed issue #1496

elena-zh commented:
Hey @fairlighteth , #1496 issue looks great to me here. I have also created #1498 issue to address low-priority issues related to the Contract icon
